Mathlib/RingTheory/PrincipalIdealDomain.lean

```diff
@@ -489,30 +489,35 @@ open Set Ideal
 variable (R) [CommRing R]
 
 /-- `nonPrincipals R` is the set of all ideals of `R` that are not principal ideals. -/
+@[deprecated "Write { I : Ideal R | ¬I.IsPrincipal } explicitely instead." (since := "2025-08-16")]
 def nonPrincipals :=
   { I : Ideal R | ¬I.IsPrincipal }
 
-theorem nonPrincipals_def {I : Ideal R} : I ∈ nonPrincipals R ↔ ¬I.IsPrincipal :=
+@[deprecated "See Ideal.nonPrincipals." (since := "2025-08-16")]
```

Comment:
I think you should add "Write { I : Ideal R | ¬I.IsPrincipal } explicitely instead." to the deprecation.